sonar-core/src/main/java/org/sonar/core/platform/ComponentContainer.java

```diff
@@ -39,6 +44,33 @@
 @ServerSide
 public class ComponentContainer implements ContainerPopulator.Container {
 
+  private static final class ExtendedDefaultPicoContainer extends DefaultPicoContainer {
+    private ExtendedDefaultPicoContainer(ComponentFactory componentFactory, LifecycleStrategy lifecycleStrategy, PicoContainer parent) {
+      super(componentFactory, lifecycleStrategy, parent);
+    }
+
+    private ExtendedDefaultPicoContainer(final ComponentFactory componentFactory, final LifecycleStrategy lifecycleStrategy, final PicoContainer parent,
+      final ComponentMonitor componentMonitor) {
+      super(componentFactory, lifecycleStrategy, parent, componentMonitor);
+    }
+
+    @Override
+    public Object getComponent(final Object componentKeyOrType, final Class<? extends Annotation> annotation) {
+      try {
+        return super.getComponent(componentKeyOrType, annotation);
+      } catch (Throwable t) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1181)